state-chain/runtime/src/chainflip/solana_elections.rs

```diff
@@ -143,16 +146,32 @@ impl OnCheckComplete<<Runtime as Chainflip>::ValidatorId> for OnCheckCompleteHoo
 #[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Encode, Decode, TypeInfo)]
 pub struct TransactionSuccessDetails {
 	pub tx_fee: u64,
+	// It is possible for a contract call to be reverted due to contract's internal error.
+	// This field is set to `true` if the contract call executed successfully without error.
+	pub contract_call_successful: bool,
 }
 
 pub struct SolanaEgressWitnessingHook;
 
 impl OnEgressSuccess<SolSignature, TransactionSuccessDetails> for SolanaEgressWitnessingHook {
 	fn on_egress_success(
 		signature: SolSignature,
-		TransactionSuccessDetails { tx_fee }: TransactionSuccessDetails,
+		TransactionSuccessDetails { tx_fee, contract_call_successful }: TransactionSuccessDetails,
```

Comment:
`contract_call_successful` is a bit missleading as it's not technically correct. I'd rename to `transaction_succesful`.

Reply:
Yep renamed.